flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/FlinkMasterDeploymentDecorator.java

```diff
@@ -0,0 +1,180 @@
+			.withSpec(podSpec).endTemplate()
+			.withNewSelector().addToMatchLabels(labels).endSelector().build());
+		return deployment;
+	}
+
+	private Container createJobManagerContainer(Configuration flinkConfig, String mainClass) {
+		String confDir = CliFrontend.getConfigurationDirectoryFromEnv();
+		String flinkConfDirInPod = flinkConfig.getString(KubernetesConfigOptions.FLINK_CONF_DIR);
+		boolean hasLogback = new File(confDir, Constants.CONFIG_FILE_LOGBACK_NAME).exists();
+		boolean hasLog4j = new File(confDir, Constants.CONFIG_FILE_LOG4J_NAME).exists();
+		String logDirInPod = flinkConfig.getString(KubernetesConfigOptions.FLINK_LOG_DIR);
+		String mainClassArgs = flinkConfig.getString(ENTRY_POINT_CLASS_ARGS);
+		String startCommand = KubernetesUtils.getJobManagerStartCommand(
+			flinkConfig,
+			clusterSpecification.getMasterMemoryMB(),
+			flinkConfDirInPod,
+			logDirInPod,
+			hasLogback,
+			hasLog4j,
+			mainClassArgs != null ? mainClass + " " + mainClassArgs : mainClass);
+
+		String image = flinkConfig.getString(KubernetesConfigOptions.CONTAINER_IMAGE);
+		String pullPolicy = flinkConfig.getString(
+			KubernetesConfigOptions.CONTAINER_IMAGE_PULL_POLICY);
+
+		double cpu = flinkConfig.getDouble(KubernetesConfigOptions.JOB_MANAGER_CPU);
+		Quantity jobManagerCpuQuantity = new Quantity(String.valueOf(cpu));
+		Quantity jobManagerMemoryQuantity = new Quantity(String.valueOf(
+			clusterSpecification.getMasterMemoryMB() + Constants.RESOURCE_UNIT_MB));
+
+		return new ContainerBuilder()
+			.withName(CONTAINER_NAME)
+			.withCommand(flinkConfig.getString(KubernetesConfigOptions.KUBERNETES_ENTRY_PATH))
+			.withArgs(Arrays.asList("/bin/bash", "-c", startCommand))
+			.withImage(image)
+			.withImagePullPolicy(pullPolicy)
+			.withResources(new ResourceRequirementsBuilder()
+				.addToRequests(Constants.RESOURCE_NAME_MEMORY, jobManagerMemoryQuantity)
+				.addToRequests(Constants.RESOURCE_NAME_CPU, jobManagerCpuQuantity)
+				.addToLimits(Constants.RESOURCE_NAME_MEMORY, jobManagerMemoryQuantity)
+				.addToLimits(Constants.RESOURCE_NAME_CPU, jobManagerCpuQuantity)
+				.build())
+			.withPorts(Arrays.asList(
+				new ContainerPortBuilder().withContainerPort(flinkConfig.getInteger(RestOptions.PORT)).build(),
+				new ContainerPortBuilder().withContainerPort(flinkConfig.getInteger(JobManagerOptions.PORT)).build(),
+				new ContainerPortBuilder().withContainerPort(BLOB_SERVER_PORT).build()))
+			.withVolumeMounts(new VolumeMountBuilder()
+				.withName(FLINK_CONF_VOLUME)
+				.withMountPath(new File(flinkConfDirInPod, FLINK_CONF_FILENAME).getPath())
+				.withSubPath(FLINK_CONF_FILENAME)
+				.build(), new VolumeMountBuilder()
+				.withName(FLINK_CONF_VOLUME)
+				.withMountPath(new File(flinkConfDirInPod, CONFIG_FILE_LOG4J_NAME).getPath())
+				.withSubPath(CONFIG_FILE_LOG4J_NAME)
+				.build(), new VolumeMountBuilder()
+				.withName(FLINK_CONF_VOLUME)
+				.withMountPath(new File(flinkConfDirInPod, CONFIG_FILE_LOGBACK_NAME).getPath())
```

Comment:
will pod launch fail if user doesn't want to provide logback configuration ?
same question when launching TM pods

Reply:
I will add a check here. If log4j/logback file does not exist, it will not be mounted into pod.